src/NServiceBus.Core.Tests/Serializers/XML/SerializerTests.cs

```diff
@@ -1227,10 +1227,78 @@ public void Object_property_with_primitive_or_struct_value_should_serialize_corr
             Assert.AreEqual(message.Value, ((SerializedPair)messageDeserialized[0]).Value);
         }
 
+
+        [Test]
+        public void Should_throw_exception_when_deserializing_payloads_with_types_existing_in_deployed_assemblies_but_not_pre_discovered_as_message_types()
+        {
+            var xmlWithBaseType = @"<?xml version=""1.0"" ?>
+                <Messages
+                    xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance""
+                    xmlns:xsd=""http://www.w3.org/2001/XMLSchema""
+                    xmlns=""http://tempuri.net/NServiceBus.Core.Tests.Serializers.XML""
+                    xmlns:baseType=""NServiceBus.Core.Tests.Serializers.XML.IMyBusMessage"">
+                    <FirstMessage></FirstMessage>
+                    <SecondMessage></SecondMessage>
+                    <SecondMessage></SecondMessage>
+                    <SecondMessage></SecondMessage>
+                </Messages>
+                ";
+
+            using (var stream = new MemoryStream())
+            {
+                var streamWriter = new StreamWriter(stream);
+                streamWriter.Write(xmlWithBaseType);
+                streamWriter.Flush();
+                stream.Position = 0;
+
+                var serializer = SerializerInstanceWithZeroMessageTypes();
+                var exception = Assert.Throws<Exception>(() => serializer.Deserialize(stream.ToArray()));
+
+                Assert.True(exception.Message.StartsWith("Could not determine type for node:"));
+            }
+        }
+
+        static XmlMessageSerializer SerializerInstanceWithZeroMessageTypes() => SerializerFactory.Create();
+
+        [Test]
+        public void Should_throw_exception_when_deserializing_payloads_with_nsb_types()
```

Comment:
can we add some actual reference to `ReplyOptions` so that this test indicates an issue in case we remove the used public type for some reason while not adjusting the test?